File: LeetCode Solutions/Arrays 101/Check If N and Its Double Exist/Check If N and Its Double Exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) {
        unordered_map<int, int> set;
        for (int i : arr) {
            if (set[2*i] > 0 || ((i%2 == 0) && (set[i/2] > 0))){
                return true;
            }
            set[i]++;
        }
        return false;
    }
};

File: LeetCode Solutions/Arrays 101/Check If N and Its Double Exist/Check If N and Its Double Exist.py
class Solution:
    def checkIfExist(self, arr: List[int]) -> bool:
        hashSet = set()
        for i in range(len(arr)):
            if ((arr[i]*2 in hashSet) or (arr[i]%2 == 0 and arr[i]//2 in hashSet)):
                return True
            hashSet.add(arr[i])
        return False